zeus/optimizer/batch_size/server/database/schema.py

```diff
@@ -0,0 +1,191 @@
+"""Database schema."""
+
+from __future__ import annotations
+
+import enum
+from datetime import datetime
+from typing import Optional
+from uuid import UUID
+
+from sqlalchemy import (
+    Boolean,
+    DateTime,
+    Enum,
+    Float,
+    ForeignKey,
+    ForeignKeyConstraint,
+    Integer,
+    Uuid,
+)
+from sqlalchemy.orm import DeclarativeBase, Mapped, mapped_column, relationship
+from sqlalchemy.sql import func
+from sqlalchemy.sql.sqltypes import VARCHAR
+from zeus.optimizer.batch_size.server.job.models import Stage
+
+
+class Base(DeclarativeBase):
+    """Base class for schemas."""
+
+    pass
+
+
+class Job(Base):
+    """Job table schema.
+
+    Refer to `JobState`[zeus.optimizer.batch_size.server.job.models] for attributes.
+    """
+
+    __tablename__ = "Job"
+
+    job_id: Mapped[UUID] = mapped_column(Uuid, primary_key=True)
+    default_batch_size: Mapped[int] = mapped_column(Integer, nullable=False)
+    high_is_better_metric: Mapped[bool] = mapped_column(Boolean, default=True)
+    eta_knob: Mapped[float] = mapped_column(Float, default=0.5)
+    beta_knob: Mapped[float] = mapped_column(Float, default=2.0)
+    target_metric: Mapped[float] = mapped_column(Float, default=0.5)
+    max_epochs: Mapped[int] = mapped_column(Integer, default=100)
+    num_pruning_rounds: Mapped[int] = mapped_column(Integer, default=2)
+    window_size: Mapped[int] = mapped_column(Integer, default=10)
+
+    max_power: Mapped[float] = mapped_column(Float, nullable=False)
+    number_of_gpus: Mapped[int] = mapped_column(Integer, nullable=False)
+    gpu_model: Mapped[str] = mapped_column(VARCHAR(length=30), nullable=False)
+
+    mab_prior_mean: Mapped[float] = mapped_column(Float, default=0.0)
+    mab_prior_precision: Mapped[float] = mapped_column(Float, default=0.0)
+    mab_num_exploration: Mapped[int] = mapped_column(Integer, default=2)
+    mab_seed: Mapped[Optional[int]] = mapped_column(Integer, nullable=True)
+
+    mab_random_generator_state: Mapped[Optional[str]] = mapped_column(
+        VARCHAR(length=300), nullable=True
+    )
+    exp_default_batch_size: Mapped[int] = mapped_column(Integer, nullable=False)
+
+    stage: Mapped[Stage] = mapped_column(Enum(Stage), default=Stage.Pruning)
+    min_cost: Mapped[Optional[float]] = mapped_column(Float, nullable=True)
+    min_batch_size: Mapped[int] = mapped_column(Integer, nullable=False)
+
+    batch_sizes: Mapped[list["BatchSize"]] = relationship(
+        order_by="BatchSize.batch_size.asc()",
+        back_populates="job",
+        # always fetch batch size(int) whenever we fetch the job.
+        # https://docs.sqlalchemy.org/en/14/orm/loading_relationships.html#relationship-loading-techniques
+        lazy="joined",
+    )
```

Comment:
Just confirming -- This fetches the list of all `BatchSize` rows when the corresponding `Job` is queried? The docstring says batch size (int) so I got a bit confused.

Reply:
When `Job` is queried, it fetches `BatchSize` too. However, it doesn't fetch any states related to that batch size, such as explorations, and measurements. It only fetches `job_id` and `batch_size: int`.